### ISSUE TITLE
Integrated tests with scipy.io

### ISSUE THREAD

**Issue by JosephThomasParker** [boutproject/BOUT-dev#893]
If `netCDF4` is not available and test suite falls back to `scipy.io`, then most integrated tests fail with the error:
`TypeError: Error: data/BOUT.dmp.0.nc is not a valid NetCDF 3 file`
because `scipy.io` cannot read netdf 4 files.  

Ideally, we would skip the tests and/or prompt the user to install netCDF4.  There is a warning about `scipy.io` printed here:
https://github.com/boutproject/BOUT-dev/blob/master/tools/pylib/boututils/datafile.py#L126
but I _don't_ think we should replace that with an error, because that function is used more widely.  Is there a change that can be made inside the test suite?

**Comment by ZedThree** [comment]
I think we now only support versions of NetCDF > 4.x, which implies that scipy.io is not compatible.

We should drop support for scipy.io for sure -- I couldn't get `Scientific` to work either, but the error wasn't very helpful and I didn't try very hard.

**Comment by JosephThomasParker** [comment]
Thanks Peter.  I'd be happy with that solution.  

@johnomotani and I found this problem installing BOUT++ on Marconi.  In our case, it was fine to just do `pip install --user netCDF4`.

**Comment by dschwoerer** [comment]
Probably related, a failed import from netCDF4 is hidden, as scipy.io is used (which is not working)

The issue is, I have installed netcdf4 using `pip3 install --user netcdf4` but import fails with:
```
$ echo "import netCDF4"|python3
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/marconi/home/userexternal/dschwoer/.local/lib/python3.5/site-packages/netCDF4/__init__.py", line 3, in <module>
    from ._netCDF4 import *
  File "netCDF4/_netCDF4.pyx", line 1099, in init netCDF4._netCDF4
  File "/marconi/home/userexternal/dschwoer/.local/lib/python3.5/site-packages/cftime/__init__.py", line 1, in <module>
    from ._cftime import utime, JulianDayFromDate, DateFromJulianDay
ImportError: /marconi/home/userexternal/dschwoer/.local/lib/python3.5/site-packages/cftime/_cftime.cpython-35m-x86_64-linux-gnu.so: undefined symbol: __intel_sse2_strchr
```
which would be nice to know - so it would be nice to print at least the error to stderr, if another library is tried ...

In the end I noticed that cftime was compiled with icc, but linked with gcc - which caused the error :-/

**Comment by ZedThree** [comment]
Let's only support `netCDF4`, everything else is terrible

**Comment by ZedThree** [comment]
Fixed in #1524 by dropping support for scipy/scientific
